### PR TITLE
[e2e] $rectorConfig->skip() should not skip inside different path over config

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba @staabm this is e2e test for check the config:

```php
    $rectorConfig->paths([
        __DIR__ . '/src',
    ]);

    $rectorConfig->skip([
        RemoveEmptyClassMethodRector::class => [
            __DIR__ . '/src/controllers',
        ],
    ]);

    $rectorConfig->rule(RemoveEmptyClassMethodRector::class);
```

which means, if the path is different, eg: `src/models`, it should still run, it currently show:

```diff
[WARNING] Register rules or sets in your "rector.php" config
```

which incorrect. 

Directly use `__DIR__ . '/src/controllers'` in skip paths() seems completely skip all, not sure if there is different issue, that probably need separate PR.

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba `$rectorConfig->skip()` seems trying to read `getcwd()` which somehow can't goes to `realpath()`, and if we try on getrector demo site, it goes error: ` Unable to write file '/var/www/public/rector.php'.` which strange 

https://getrector.com/demo/5871515d-0885-4df1-bd65-ef36442d5703

**Comment by samsonasik** [comment]
@TomasVotruba it seems bug on `ContainerMemento` class, the service is `unset()` when using `skip()` even when define path over the skip class:

- https://github.com/rectorphp/rector-src/blob/8b07c8e7de96a3b304b5409f1ef3415a513fee91/src/DependencyInjection/Laravel/ContainerMemento.php#L28

**Comment by samsonasik** [comment]
@TomasVotruba @staabm fixed 🎉 , the forgetting service on `skip()` should only when path is not specified https://github.com/rectorphp/rector-src/pull/4882/commits/fec890d3be6626af9527946dd13488aa2d5eb92e

**Comment by samsonasik** [comment]
All checks have passed 🎉 🎉 🎉 @TomasVotruba I think it is ready 🎉 🎉 🎉 🎉 🎉

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you 😊👍

File: src/DependencyInjection/LazyContainerFactory.php [hunk 1]
```diff
@@ -729,9 +729,12 @@ static function (
...
-                ContainerMemento::forgetService($container, $skippedClass);
+            $skippedElements = $skippedClassResolver->resolve();
+            foreach ($skippedElements as $skippedClass => $path) {
+                // completely forget the Rector rule only when no path specified
+                if ($path === null) {
+                    ContainerMemento::forgetService($container, $skippedClass);
+                }
             }
 
             $hasForgotten = true;
```
Comment (staabm) [inline src/DependencyInjection/LazyContainerFactory.php:740]: is this flag still correct then?

File: src/DependencyInjection/LazyContainerFactory.php [hunk 2]
```diff
@@ -729,9 +729,12 @@ static function (
...
 
             $skippedClassResolver = new SkippedClassResolver();
-            $skippedClasses = array_keys($skippedClassResolver->resolve());
-            foreach ($skippedClasses as $skippedClass) {
-                ContainerMemento::forgetService($container, $skippedClass);
+            $skippedElements = $skippedClassResolver->resolve();
+            foreach ($skippedElements as $skippedClass => $path) {
+                // completely forget the Rector rule only when no path specified
+                if ($path === null) {
+                    ContainerMemento::forgetService($container, $skippedClass);
```
Comment (staabm) [inline src/DependencyInjection/LazyContainerFactory.php:736]: could you give me an idea why we need to drop classes from the container when rules are skipped?

**Comment by staabm** [comment]
thanks for the fix!

File: src/DependencyInjection/LazyContainerFactory.php (hunk 2 above)
Comment (samsonasik) [reply src/DependencyInjection/LazyContainerFactory.php:736]: This @TomasVotruba blog post may be the reason https://tomasvotruba.com/blog/removing-service-from-laravel-container-is-not-that-easy

File: src/DependencyInjection/LazyContainerFactory.php (hunk 1 above)
Comment (samsonasik) [reply src/DependencyInjection/LazyContainerFactory.php:740]: That seems used on line 727 to run once on phpunit

**Pull request merged**